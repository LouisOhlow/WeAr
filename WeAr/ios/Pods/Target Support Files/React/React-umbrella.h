#ifdef __OBJC__
#import <UIKit/UIKit.h>
#else
#ifndef FOUNDATION_EXPORT
#if defined(__cplusplus)
#define FOUNDATION_EXPORT extern "C"
#else
#define FOUNDATION_EXPORT extern
#endif
#endif
#endif

#import "React/Base/RCTAssert.h"
#import "React/Base/RCTBridge.h"
#import "React/Base/RCTBridgeDelegate.h"
#import "React/Base/RCTBridgeModule.h"
#import "React/Base/RCTCache.h"
#import "React/Base/RCTConvert.h"
#import "React/Base/RCTDefines.h"
#import "React/Base/RCTEventDispatcher.h"
#import "React/Base/RCTFPSGraph.h"
#import "React/Base/RCTFrameUpdate.h"
#import "React/Base/RCTInvalidating.h"
#import "React/Base/RCTJavaScriptExecutor.h"
#import "React/Base/RCTJavaScriptLoader.h"
#import "React/Base/RCTJSMethodRegistrar.h"
#import "React/Base/RCTKeyboardObserver.h"
#import "React/Base/RCTKeyCommands.h"
#import "React/Base/RCTLog.h"
#import "React/Base/RCTModuleData.h"
#import "React/Base/RCTModuleMap.h"
#import "React/Base/RCTModuleMethod.h"
#import "React/Base/RCTPerformanceLogger.h"
#import "React/Base/RCTPerfStats.h"
#import "React/Base/RCTProfile.h"
#import "React/Base/RCTRootView.h"
#import "React/Base/RCTSparseArray.h"
#import "React/Base/RCTTouchHandler.h"
#import "React/Base/RCTURLRequestDelegate.h"
#import "React/Base/RCTURLRequestHandler.h"
#import "React/Base/RCTUtils.h"
#import "React/Executors/RCTContextExecutor.h"
#import "React/Executors/RCTWebViewExecutor.h"
#import "React/Layout/Layout.h"
#import "React/Modules/RCTAccessibilityManager.h"
#import "React/Modules/RCTAlertManager.h"
#import "React/Modules/RCTAppState.h"
#import "React/Modules/RCTAsyncLocalStorage.h"
#import "React/Modules/RCTDevLoadingView.h"
#import "React/Modules/RCTDevMenu.h"
#import "React/Modules/RCTExceptionsManager.h"
#import "React/Modules/RCTPointAnnotation.h"
#import "React/Modules/RCTRedBox.h"
#import "React/Modules/RCTSourceCode.h"
#import "React/Modules/RCTStatusBarManager.h"
#import "React/Modules/RCTTiming.h"
#import "React/Modules/RCTUIManager.h"
#import "React/Views/RCTActivityIndicatorViewManager.h"
#import "React/Views/RCTAnimationType.h"
#import "React/Views/RCTAutoInsetsProtocol.h"
#import "React/Views/RCTBorderDrawing.h"
#import "React/Views/RCTComponent.h"
#import "React/Views/RCTComponentData.h"
#import "React/Views/RCTConvert+CoreLocation.h"
#import "React/Views/RCTConvert+MapKit.h"
#import "React/Views/RCTDatePickerManager.h"
#import "React/Views/RCTMap.h"
#import "React/Views/RCTMapManager.h"
#import "React/Views/RCTModalHostView.h"
#import "React/Views/RCTModalHostViewController.h"
#import "React/Views/RCTModalHostViewManager.h"
#import "React/Views/RCTNavigator.h"
#import "React/Views/RCTNavigatorManager.h"
#import "React/Views/RCTNavItem.h"
#import "React/Views/RCTNavItemManager.h"
#import "React/Views/RCTPicker.h"
#import "React/Views/RCTPickerManager.h"
#import "React/Views/RCTPointerEvents.h"
#import "React/Views/RCTProgressViewManager.h"
#import "React/Views/RCTScrollableProtocol.h"
#import "React/Views/RCTScrollView.h"
#import "React/Views/RCTScrollViewManager.h"
#import "React/Views/RCTSegmentedControl.h"
#import "React/Views/RCTSegmentedControlManager.h"
#import "React/Views/RCTShadowView.h"
#import "React/Views/RCTSlider.h"
#import "React/Views/RCTSliderManager.h"
#import "React/Views/RCTSwitch.h"
#import "React/Views/RCTSwitchManager.h"
#import "React/Views/RCTTabBar.h"
#import "React/Views/RCTTabBarItem.h"
#import "React/Views/RCTTabBarItemManager.h"
#import "React/Views/RCTTabBarManager.h"
#import "React/Views/RCTTextDecorationLineType.h"
#import "React/Views/RCTView.h"
#import "React/Views/RCTViewControllerProtocol.h"
#import "React/Views/RCTViewManager.h"
#import "React/Views/RCTWebView.h"
#import "React/Views/RCTWebViewManager.h"
#import "React/Views/RCTWrapperViewController.h"
#import "React/Views/UIView+Private.h"
#import "React/Views/UIView+React.h"

FOUNDATION_EXPORT double ReactVersionNumber;
FOUNDATION_EXPORT const unsigned char ReactVersionString[];

