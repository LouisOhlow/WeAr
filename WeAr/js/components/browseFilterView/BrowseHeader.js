import React from 'react';
import { View, StyleSheet } from 'react-native';
import NavigationButton from '../navigation/NavigationButton';
import Headline2 from '../basics/Headline2';
import COLORS from '../../drawables/colors';

/**
 * displays the Headline and the button to move back to the CameraView component
 */
const BrowseHeader = (props) => {
  const { navigate } = props;
  return (
    <View style={styles.container}>
      <View style={styles.button}>
        <NavigationButton style={styles.button} onPress={() => navigate()} direction="up" />
      </View>
      <View style={styles.header}>
        <Headline2 text="CHOOSE FILTER" />
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.background,
    width: '100%',
    height: '15%',
  },
  button: {
    alignSelf: 'center',
    top: 20,
  },
  header: {
    alignSelf: 'center',
    top: 20,
  },
});

export default BrowseHeader;
