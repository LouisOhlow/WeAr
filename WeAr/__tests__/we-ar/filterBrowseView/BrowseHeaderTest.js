import React from 'react';
import renderer from 'react-test-renderer';
import BrowseHeader from '../../../js/components/browseFilterView/BrowseHeader';

test('snapshot testing the browse header', () => {
        const tree = renderer
          .create(<BrowseHeader navigate={() => {}} />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });
