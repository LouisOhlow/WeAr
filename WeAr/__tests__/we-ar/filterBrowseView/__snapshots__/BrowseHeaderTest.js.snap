// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing the browse header 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#222222",
      "height": "15%",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "top": 20,
      }
    }
  >
    <View>
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 70,
            "justifyContent": "center",
            "opacity": 1,
            "width": 70,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../js/drawables/arrow_button.png",
            }
          }
          style={
            Object {
              "height": 70,
              "resizeMode": "stretch",
              "transform": Array [
                Object {
                  "rotate": "270deg",
                },
              ],
              "width": 30,
            }
          }
        />
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "top": 20,
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 30,
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        CHOOSE FILTER
      </Text>
    </View>
  </View>
</View>
`;
